File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        dic = {}
        bgnumbers = []
        longest = 0
        #convert to dict
        for num in nums:
            dic[num] = 1
        #find the begin number
        for num in nums:
            if not dic.get(num-1): bgnumbers.append(num)
        for bg in bgnumbers:
            l = 1
            temp = bg
            while dic.get(temp+1):
                print(temp)
                l+=1
                temp+=1
            if l > longest: longest = l
        return longest
        
                
        

File: Data Structures & Algorithms/maximum-subarray/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    maxSubArray(nums: number[]): number {
        let cursum = nums[0], max = nums[0];
        for (let i=1; i < nums.length; i++) {
            cursum+= nums[i];
            if(cursum < nums[i]) {
                cursum = nums[i];
            }

            if( cursum > max ) {
                max = cursum;
            }
        }

        return max;
    }
}
